araproc/analysis/rpr.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from scipy.ndimage import uniform_filter1d
+from araproc.framework import waveform_utilities as wfu
+from araproc.analysis import snr
+
+def get_rpr(waveform):
+    """
+    Computes the RPR (Ratio of Peak to RMS) value, similar to SNR, for the given waveform.
+
+    Parameters
+    ----------
+    waveform: TGraph
+        A TGraph of the waveform containing time and voltage values.
+
+    Returns
+    -------
+    rpr_val: float
+        The RPR value for the waveform, calculated as the ratio of the maximum voltage 
+        (after smoothing) to the standard deviation of the waveform parts with the least RMS.
+    """
+    # Extract time and voltage arrays from the waveform
+    channel_time, channel_wf = wfu.tgraph_to_arrays(waveform)
+    wf_len = len(channel_wf)
+    
+    # Square the waveform data for further processing
+    channel_wf = channel_wf ** 2
+
+    # Calculate the smoothing window size based on sampling rate
+    dt =  wfu.get_dt_and_sampling_rate(channel_time)[0]
+    sum_win = 25  # Smoothing window in ns
+    sum_win_idx = int(np.round(sum_win / dt))  # Convert window size to sample points
+    channel_wf = np.sqrt(uniform_filter1d(channel_wf, size=sum_win_idx, mode='constant'))
+
+    # Find the maximum value of the smoothed waveform
+    max_bin = np.nanargmax(channel_wf)
+    max_val = channel_wf[max_bin]
+
+    # Calculate noise sigma: standard deviation of the four parts with the least RMS
```

Comment:
Sorry, I forgot that `snr.get_min_segmented_rms`  didn't accept an array previously. It will now if you do a git pull, so I think we can replace this block by `sigma = snr.get_min_segmented_rms(channel_wf)`.

Reply:
Hmmm, Okay it was not accepting  the channel_wf before, I can call snr function then.